internal/gps/result.go

```diff
@@ -43,38 +45,65 @@ type solution struct {
 	solv Solver
 }
 
-// WriteDepTree takes a basedir and a Lock, and exports all the projects
-// listed in the lock to the appropriate target location within the basedir.
+// WriteDepTree takes a baseDir and a Lock, and exports all the projects
+// listed in the lock to the appropriate target location within baseDir.
 //
-// If the goal is to populate a vendor directory, basedir should be the absolute
+// If the goal is to populate a vendor directory, baseDir should be the absolute
 // path to that vendor directory, not its parent (a project root, typically).
 //
-// It requires a SourceManager to do the work, and takes a flag indicating
-// whether or not to strip vendor directories contained in the exported
-// dependencies.
-func WriteDepTree(basedir string, l Lock, sm SourceManager, sv bool, logger *log.Logger) error {
+// It requires a SourceManager to do the work, and takes a PruneOptions
+// indicating the pruning options required for the exported dependencies.
+func WriteDepTree(baseDir string, l Lock, sm SourceManager, prune PruneOptions, logger *log.Logger) error {
+	if baseDir == "" {
+		return errors.New("must provide a non-empty baseDir")
+	}
 	if l == nil {
-		return fmt.Errorf("must provide non-nil Lock to WriteDepTree")
+		return errors.New("must provide a non-nil Lock to WriteDepTree")
+	}
+	if sm == nil {
+		return errors.New("must provide a non-nil SourceManager to WriteDepTree")
 	}
 
-	err := os.MkdirAll(basedir, 0777)
-	if err != nil {
+	if err := os.MkdirAll(baseDir, 0777); err != nil {
 		return err
 	}
 
-	// TODO(sdboyer) parallelize
+	var wg sync.WaitGroup
```

Comment:
also doing the parallelizing here? in general, would prefer to keep down the number of major changes in a single PR.